doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,146 @@
+
+- Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx output that sits at the Plutus script address. It must have a datum hash, otherwise, it is unspendable.
+- The reference transaction input containing the corresponding Plutus script. We must create the transaction output containing the reference Plutus script.
+
+#### Creating the `Required Redeemer` Plutus spending script
+
+The plutus-example executable will automagically generate several Plutus scripts in the CLI-compatible text envelope format.
+
+Run the following commands:
+
+```bash
+git clone git@github.com:input-output-hk/plutus-apps.git
+
+cd plutus-apps/plutus-example
+
+cabal run exe:plutus-example
+```
+
+This will output `required-redeemer.plutus` in the `generated-plutus-scripts/v2` dir.
+
+#### Setting up a local Babbage node cluster
+
+There is a convenient script that will set up a Babbage cluster immediately on your local machine.
+
+Run the following command:
+
+```bash
+cabal install cardano-cli
+cabal install cardano-node
+./scripts/babbage/mkfiles.sh
+```
+
+To start your babbage cluster you need to run the `example/run/all.sh` shell script.
+The remainder of this guide briefly talks about the [shell script example](scripts/plutus/example-reference-script-usage.sh) that automatically does everything for you.
+
+#### Creating a reference script at a transaction output and
+#### sending ADA to the script address (with a datum)
+
+In order to use a reference script we must first create said reference script at a particular address. Note you cannot create a reference script at the same correspnding script address, this will result in a failure when trying to spend the transaction input at the script address.
+
+```bash
+cardano-cli transaction build \
+  --babbage-era \
+  --cardano-mode \
+  --testnet-magic 42 \
+  --change-address "$utxoaddr" \
+  --tx-in "$txin" \
+  --tx-out "$utxoaddr+$lovelace" \
+  --tx-out "$plutusscriptaddr+$lovelace" \
+  --tx-out-datum-hash "$scriptdatumhash" \
+  --tx-out "$dummyaddress+$lovelaceattxindiv3" \
+  --reference-script-file "$plutusscriptinuse" \
+  --protocol-params-file "$WORK/pparams.json" \
+  --out-file "$WORK/create-datum-output.body"
+```
+
+There are a couple things happening in the following `build` command.
+Firstly we are sending ADA to the plutus script address along with a datum. This is reflected in the following lines:
+
+```bash
+...
+--tx-out "$plutusscriptaddr+$lovelace" \
+--tx-out-datum-hash "$scriptdatumhash" \
+...
+```
+
+We have seen this before in the [plutus-spending-script-example.md](doc/reference/plutus/plutus-spending-script-example.md).
+
+Secondly we are creating a reference script at an address, which cannot be the same as the reference script address, of our choosing in the following lines:
```

Comment:
Secondly, we are creating a reference script at an address, which cannot be the same as the reference script address of our choice: